arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -1085,6 +1142,53 @@ bool IResearchAnalyzerFeature::loadConfiguration() {
   return FEATURE_NAME;
 }
 
+/*static*/ std::string IResearchAnalyzerFeature::normalize( // normalize name
+  irs::string_ref const& name, // analyzer name
+  TRI_vocbase_t const& activeVocbase, // fallback vocbase if not part of name
+  TRI_vocbase_t const& systemVocbase, // the system vocbase for use with empty prefix
+  bool expandVocbasePrefix /*= true*/ // use full vocbase name as prefix for active/system v.s. EMPTY/'::'
+) {
+  if (IDENTITY_ANALYZER_NAME == name) {
+    return name; // special case for the 'identity' analyzer which is a singleton
+  }
+
+  // FIXME TODO remove once lagacy analyzers will be added to each vocbase via an upgrade step and JavaScript tests updated accordingly
+  {
+    static std::unordered_set<std::string> legacyAnalyzers = {
```

Comment:
can we make this set const?
```suggestion
    static std::unordered_set<std::string> const legacyAnalyzers = {
```

Reply:
it'll be removed in the next pull-request